modules/tinymce/src/core/main/ts/fmt/MatchFormat.ts

```diff
@@ -67,65 +67,65 @@ const matchName = (dom: DOMUtils, node: Node, format) => {
   }
 };
 
-const matchItems = (dom: DOMUtils, node: Node, format, itemName: string, similar: boolean, vars: FormatVars) => {
-  let key, value;
+const matchItems = (dom: DOMUtils, node: Node, format: Format, itemName: string, similar: boolean, vars: FormatVars): boolean => {
   const items = format[itemName];
-  let i;
 
   // Custom match
   if (format.onmatch) {
-    return format.onmatch(node, format, itemName);
+    const onMatch: (node: Node, format: Format, name: string) => boolean = format.onmatch;
```

Comment:
The reason this cast is here is because `format.onmatch` technically can't take `Format` as its second argument. The `.onmatch` of a `SelectorFormat` can only accept a `SelectorFormat`, the `.onmatch` of a `BlockFormat` can only accept a `BlockFormat`, etc etc. The cast is safe, because we only pass the format itself into this `onmatch` function, but unfortunately I'm not aware of any way to skip the cast.